react/hooks/useSimulation.js

```diff
@@ -0,0 +1,79 @@
+import { useMemo } from 'react'
+import { useQuery } from 'react-apollo'
+import itemsWithSimulationQuery from 'vtex.store-resources/QueryItemsWithSimulation'
+
+import useIsPriceAsync from './useIsPriceAsync'
+import clone from '../utils/clone'
+
+const mergeSellers = (sellerA, sellerB) => {
+  sellerA.commertialOffer = {
+    ...sellerA.commertialOffer,
+    ...sellerB.commertialOffer,
+  }
+
+  // Deal with withoutPriceFulfillment cases
+  if (sellerA.commertialOffer.Price === 0) {
+    sellerA.commertialOffer.AvailableQuantity = 0
+  }
+
+  return sellerA
+}
+
+const useSimulation = ({ product, inView, onComplete, onError }) => {
+  const { isPriceAsync } = useIsPriceAsync()
+
+  const simulationItemsInput = useMemo(
+    () =>
+      product.items.map((item) => ({
+        itemId: item.itemId,
+        sellers: item.sellers.map((seller) => ({
+          sellerId: seller.sellerId,
+        })),
+      })),
+    [product]
+  )
+
+  useQuery(itemsWithSimulationQuery, {
+    variables: {
+      items: simulationItemsInput,
+    },
+    skip: !isPriceAsync || !inView,
+    ssr: false,
+    onError,
+    onCompleted: (response) => {
+      const simulationItems = response.itemsWithSimulation
+
+      const mergedProduct = clone(product)
```

Comment:
We do we have to merge, can't we just update the whole product data?

Reply:
Unfortunately, the simulation API doesn't return the whole item,  so I need to do this merge 😢